core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/dictionary/DictionaryMatcherOpExec.scala

```diff
@@ -0,0 +1,149 @@
+package edu.uci.ics.texera.workflow.operators.dictionary
+
+import edu.uci.ics.amber.engine.common.InputExhausted
+import edu.uci.ics.amber.engine.common.virtualidentity.LinkIdentity
+import edu.uci.ics.texera.workflow.common.operators.OperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.{AttributeType, OperatorSchemaInfo}
+import org.apache.lucene.analysis.Analyzer
+import org.apache.lucene.analysis.en.EnglishAnalyzer
+import org.apache.lucene.analysis.tokenattributes.OffsetAttribute
+import scala.collection.mutable.Set
+import java.io.StringReader
+import scala.collection.Iterator
+import scala.collection.mutable.ListBuffer
+import scala.util.Either
+
+class DictionaryMatcherOpExec(
+    opDesc: DictionaryMatcherOpDesc,
+    operatorSchemaInfo: OperatorSchemaInfo
+) extends OperatorExecutor {
+
+  // this is needed for the matching types Phrase and Conjunction
+  var tokenizedDictionaryEntries: ListBuffer[Set[String]] = _
+  // this is needed for the simple Scan matching type
+  var dictionaryEntries: List[String] = _
+  var luceneAnalyzer: Analyzer = _
+
+  /**
+    * first prepare the dictionary by splitting the values using a comma delimiter then tokenize the splitted values
+    */
+  override def open(): Unit = {
+    // create the dictionary by splitting the values first
+    dictionaryEntries = this.opDesc.dictionary.split(",").toList.map(_.toLowerCase)
+    // then tokenize each entry
+    this.luceneAnalyzer = new EnglishAnalyzer
+    tokenizedDictionaryEntries = ListBuffer[Set[String]]()
+    tokenizeDictionary();
+  }
+
+  override def close(): Unit = {
+    tokenizedDictionaryEntries = null
+    dictionaryEntries = null
+    luceneAnalyzer = null
+  }
+
+  /**
+    * use LuceneAnalyzer to tokenize the dictionary
+    */
+  def tokenizeDictionary(): Unit = {
+    for (text <- dictionaryEntries) {
+      tokenizedDictionaryEntries += tokenize(text)
+    }
+  }
+
+  /**
+    * decide if a tuple matches a dictionary after it is tokenized based on {@link MatchingType}
+    * @param tuple
+    * @return
+    */
+  def isTupleInDictionary(tuple: Tuple): Boolean = {
+    // first, tokenize the tuple
+    val tokenizedTuple = tokenize(tuple.getField(this.opDesc.attribute))
```

Comment:
I assume the tokenization of input tuple is not needed in case of the scan match type?  (same comment for tokenization of dictionary)